Scatt.java

```diff
@@ -24,5 +26,28 @@
      */
     public static void main(String[] args)
     {
+        System.out.println("Please enter the folder name: ");
+        String dirName = System.console().readLine();
+        File directory = new File(dirName);
+        Boolean isValid = readValidDirectory(directory);
+    }
+
+    /**
+     * Check to see if folder is valid.
+     *
+     * @param dir directory file object
+     * @return true or false
+     */
+    public static Boolean readValidDirectory(File dir)
+    {
+        if (dir.exists() || dir.isDirectory())
```

Comment:
Should this be && - don't we want to ensure that the directory exists AND that it is a directory?